src/Settings.ts

```diff
@@ -0,0 +1,107 @@
+import * as fs from "fs";
+import * as path from "path";
+import { EnvironmentMode } from "./services/DeploymentService";
+
+const devMode = process.env.RUN_MODE && (<string>process.env.RUN_MODE).trim() === "dev";
+if (devMode) {
+    console.log("Running in dev mode");
+}
+
+const projectPath = path.resolve(__dirname, devMode ? "../" : "../../../");
+const pkg: any = require(path.resolve(projectPath, "package.json"));
+
+export interface MinimalSettings {
+    appName: string;
+    key: string;
+    user: string;
+    password: string;
+    projectId: string;
+}
+export interface Settings extends MinimalSettings {
+    apiUrl: string;
+    branchName: string;
+    environment: EnvironmentMode;
+    teamServerUrl: string;
+    testProjectName: string;
+    widget: {
+        name: string;
+        version: string;
+    };
+    folder: {
+        build: string;
+        dist: string;
+        release: string;
+    };
+}
+
+export const defaultSettings: PartialSettings = {
+    branchName: "trunk",
+    environment: "Sandbox",
+    testProjectName: "TestProject.mpk"
+};
+
+export type Partial<T> = {
+    [P in keyof T]?: T[P];
+};
+
+export type PartialSettings = Partial<Settings>;
```

Comment:
**Prefer** it stays in it's original form `Partial<Settings>` and export only Settings... difference is just two angle brackets :)